### ISSUE TITLE
Problems Error on Build & Flash The ESP32 under Eclipse IDE

### ISSUE THREAD

**Issue by apriliadina** [espressif/esp-idf#2129]
Hi, 

I'm trying to build and flash the example project of hello world and many others in ESP32.
I'm using the Windows 8.1, the latest update of msys2 and eclipse IDE cpp Photon but when I tried to build and flash the program in Eclipse I keep getting this kind of error messages. But strangely, I can make flash from the msys32 built in command window and Arduino IDE.
I've followed all the step written in https://esp-idf.readthedocs.io/en/latest/get-started/eclipse-setup-windows.html#eclipse-windows-setup 

Here is the full trace that written in the console program:

Target 'component-main-build' responsible for 'C:/pci_internship/msys32/home/lennovo/esp/hello_world/build/bootloader/main/libmain.a'
ERROR:root:code for hash md5 was not found.
Traceback (most recent call last):
  File "C:\pci_internship\msys32\mingw32\lib\python2.7/hashlib.py", line 147, in <module>
    globals()[__func_name] = __get_hash(__func_name)
  File "C:\pci_internship\msys32\mingw32\lib\python2.7/hashlib.py", line 97, in __get_builtin_constructor
    raise ValueError('unsupported hash type ' + name)
ValueError: unsupported hash type md5
ERROR:root:code for hash sha1 was not found.
Traceback (most recent call last):
  File "C:\pci_internship\msys32\mingw32\lib\python2.7/hashlib.py", line 147, in <module>
    globals()[__func_name] = __get_hash(__func_name)
  File "C:\pci_internship\msys32\mingw32\lib\python2.7/hashlib.py", line 97, in __get_builtin_constructor
    raise ValueError('unsupported hash type ' + name)
ValueError: unsupported hash type sha1
ERROR:root:code for hash sha224 was not found.
Traceback (most recent call last):
  File "C:\pci_internship\msys32\mingw32\lib\python2.7/hashlib.py", line 147, in <module>
    globals()[__func_name] = __get_hash(__func_name)
  File "C:\pci_internship\msys32\mingw32\lib\python2.7/hashlib.py", line 97, in __get_builtin_constructor
    raise ValueError('unsupported hash type ' + name)
ValueError: unsupported hash type sha224
ERROR:root:code for hash sha256 was not found.
Traceback (most recent call last):
  File "C:\pci_internship\msys32\mingw32\lib\python2.7/hashlib.py", line 147, in <module>
    globals()[__func_name] = __get_hash(__func_name)
  File "C:\pci_internship\msys32\mingw32\lib\python2.7/hashlib.py", line 97, in __get_builtin_constructor
    raise ValueError('unsupported hash type ' + name)
ValueError: unsupported hash type sha256
ERROR:root:code for hash sha384 was not found.
Traceback (most recent call last):
  File "C:\pci_internship\msys32\mingw32\lib\python2.7/hashlib.py", line 147, in <module>
    globals()[__func_name] = __get_hash(__func_name)
  File "C:\pci_internship\msys32\mingw32\lib\python2.7/hashlib.py", line 97, in __get_builtin_constructor
    raise ValueError('unsupported hash type ' + name)
ValueError: unsupported hash type sha384
ERROR:root:code for hash sha512 was not found.
Traceback (most recent call last):
  File "C:\pci_internship\msys32\mingw32\lib\python2.7/hashlib.py", line 147, in <module>
    globals()[__func_name] = __get_hash(__func_name)
  File "C:\pci_internship\msys32\mingw32\lib\python2.7/hashlib.py", line 97, in __get_builtin_constructor
    raise ValueError('unsupported hash type ' + name)
ValueError: unsupported hash type sha512
Traceback (most recent call last):
  File "C:/pci_internship/msys32/home/lennovo/esp/esp-idf/components/esptool_py/esptool/esptool.py", line 2637, in <module>
    _main()
  File "C:/pci_internship/msys32/home/lennovo/esp/esp-idf/components/esptool_py/esptool/esptool.py", line 2630, in _main
    main()
  File "C:/pci_internship/msys32/home/lennovo/esp/esp-idf/components/esptool_py/esptool/esptool.py", line 2403, in main
    operation_func(args)
  File "C:/pci_internship/msys32/home/lennovo/esp/esp-idf/components/esptool_py/esptool/esptool.py", line 2009, in elf2image
    image.save(args.output)
  File "C:/pci_internship/msys32/home/lennovo/esp/esp-idf/components/esptool_py/esptool/esptool.py", line 1513, in save
    digest = hashlib.sha256()
AttributeError: 'module' object has no attribute 'sha256'
**make[1]: *** [/home/lennovo/esp/esp-idf/components/esptool_py/Makefile.projbuild:51: /home/lennovo/esp/hello_world/build/bootloader/bootloader.bin] Error 1
make: *** [/home/lennovo/esp/esp-idf/components/bootloader/Makefile.projbuild:41: /home/lennovo/esp/hello_world/build/bootloader/bootloader.bin] Error 2**
xtensa-esp32-elf-gcc -nostdlib -u call_user_start_cpu0  -Wl,--gc-sections -Wl,-static -Wl,--start-group  -LC:/pci_internship/msys32/home/lennovo/esp/hello_world/build/bootloader/bootloader_support -lbootloader_support -LC:/pci_internship/msys32/home/lennovo/esp/hello_world/build/bootloader/log -llog -LC:/pci_internship/msys32/home/lennovo/esp/hello_world/build/bootloader/spi_flash -lspi_flash -LC:/pci_internship/msys32/home/lennovo/esp/hello_world/build/bootloader/micro-ecc -lmicro-ecc -LC:/pci_internship/msys32/home/lennovo/esp/hello_world/build/bootloader/soc -lsoc -LC:/pci_internship/msys32/home/lennovo/esp/hello_world/build/bootloader/main -lmain -L C:/pci_internship/msys32/home/lennovo/esp/esp-idf/components/bootloader/subproject/main -T esp32.bootloader.ld -T C:/pci_internship/msys32/home/lennovo/esp/esp-idf/components/esp32/ld/esp32.rom.ld -T C:/pci_internship/msys32/home/lennovo/esp/esp-idf/components/esp32/ld/esp32.rom.spiram_incompatible_fns.ld -T C:/pci_internship/msys32/home/lennovo/esp/esp-idf/components/esp32/ld/esp32.peripherals.ld -T esp32.bootloader.rom.ld -lgcc -lstdc++ -lgcov -Wl,--end-group -Wl,-EL -o C:/pci_internship/msys32/home/lennovo/esp/hello_world/build/bootloader/bootloader.elf -Wl,-Map=C:/pci_internship/msys32/home/lennovo/esp/hello_world/build/bootloader/bootloader.map
python C:/pci_internship/msys32/home/lennovo/esp/esp-idf/components/esptool_py/esptool/esptool.py --chip esp32 elf2image --flash_mode "dio" --flash_freq "40m" --flash_size "2MB"  -o /home/lennovo/esp/hello_world/build/bootloader/bootloader.bin C:/pci_internship/msys32/home/lennovo/esp/hello_world/build/bootloader/bootloader.elf
esptool.py v2.3.1
make[1]: Leaving directory 'C:/pci_internship/msys32/home/lennovo/esp/esp-idf/components/bootloader/subproject'

Anyone has the same problem like me? I really need your help to fix this, thank you.

**Comment by projectgus** [comment]
Hi @apriliadina ,

Sorry noone got back to you earlier. Are you closing the issue because you solved the problem? If so, could you please leave any tip for someone who finds this issue in the future?

Thanks,

Angus

**Comment by apriliadina** [comment]
Hi, Angus

Yeah, It's okay. I'm not really solved the problems, and I've tried to do
anything with my Eclipse IDE, toolchain, and even the python since last
weeks. So, because of my lack time, and I really need to use my ESP32 in
Eclipse IDE for my project, I decided to reinstall my windows, and upgrade
it to windows 10. Suprisingly it is works. But, I still don't know the
right way to solve that problem.

Thanks,

Dina

2018-07-03 10:36 GMT+07:00 Angus Gratton <notifications@github.com>:

**Comment by asmitajha** [comment]
Hi
I was able to build and flash successfully using eclipse IDE. But the problem is that when I checked with blinking LED example, the LED was not blinking on the board. I also tried by changing its GPIO in menuconfig. But when i flashed using msys command prompt using same examples and menuconfig the led was blinking on the board. So i want to ask that why is it not blinking using when flashed with eclipse?﻿
